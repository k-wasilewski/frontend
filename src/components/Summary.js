import React, {Component} from 'react';
import '../css/App.css';

class Summary extends Component {

    render() {
        return (
            <div className="main">
                podsumowanie
            </div>
        );
    }
}

export default Summary;
