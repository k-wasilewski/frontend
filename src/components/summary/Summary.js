import React, {Component} from 'react';
import '../../css/App.css';
import axios from "axios";
import TranslateItems from "../../func/TranslateItems";
import {connect} from "react-redux";
import {Redirect} from "react-router-dom";

export class Summary extends Component {
    constructor(props) {
        super(props);
        this.state = {
            list: '',
            error: null
        };

        this.doGetList = this.doGetList.bind(this);
    };

    doGetList() {
        const axiosConfig = {
            headers: {
                "Access-Control-Allow-Origin": "*",
                "Authorization": `Bearer ${localStorage.getItem('token')}`
            }
        };

        axios.get(`http://localhost:8081/auth/list?username=${this.props.username}`,
            axiosConfig
        ).then(resp => {
            this.setState({list: resp.data});
        }).catch(() => {
            this.setState({error: 'Błąd serwera'});
        });
    }

    formatItems = (items) => {
        let itemsTransformed = [];

        TranslateItems(items)
        items.forEach(function(item) {
            itemsTransformed.push(
                <li key={itemsTransformed.length}>
                    Kolor: {item.color}<br/>
                    Rozmiar: {item.size}<br/>
                </li>
            );
        })

        return itemsTransformed
    };

    formatList = (list) => {
        let transformedList = [];

        for (let i=0; i<list.length; i++) {
            const name = list[i].name;
            const age = list[i].age;
            const created = list[i].created;

            const items = list[i].items;
            let itemsTransformed;
            if (items!==undefined) itemsTransformed = this.formatItems(items);

            const key = name+age+created;
            transformedList.push((
                <ul key={key}>
                    <li id='itemList'>
                        Imię: {name}<br/>
                        Wiek: {age}<br/>
                        <button className='showElems' onClick={this.showOrderList}>Elementy</button> <br/>
                        <div className='orderItems hidden'>
                            <ol start='1'>{itemsTransformed}</ol><br/>
                        </div>
                        Data utworzenia: {created}
                    </li>
                </ul>
            ));
        }

        return transformedList;
    };

    showOrderList(event) {
        const $this = event.target;
        const list = $this.parentElement.getElementsByClassName('orderItems')[0];
        list.classList.toggle('hidden');
    };

    render() {
        let list = this.state.list;
        list = this.formatList(list);

        if (!this.props.username) return (
            <Redirect to='/login' />
        );
        else return (
            <div className="main">
                <span style={{color: 'red'}}>{this.state.error}</span>
                <div className='summary'>
                    <h2>Podsumowanie</h2>
                    <div className='col1'>
                        {list}
                    </div>
                    <div className='col2'>
                        <button onClick={this.doGetList}>Szukaj</button>
                    </div>
                </div>
            </div>
        );
    };
}

function mapStateToProps(state) {
    return {
        username: state.setUsernameReducer.username
    };
}

export default connect(mapStateToProps, null)(Summary);
