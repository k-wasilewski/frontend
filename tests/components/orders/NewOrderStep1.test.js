import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../../src/redux/store";
import {Provider} from "react-redux";
import {configure, shallow} from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import ConnectedNewOrderStep1, { NewOrderStep1 } from "../../../src/components/orders/NewOrderStep1";
import {Summary} from "../../../src/components/summary/Summary";

describe("NewOrderStep1 specification", () => {
    it('should render a form with two inputs', () => {
        const component = renderer.create(
            <Provider store={store}>
                <ConnectedNewOrderStep1 />
            </Provider>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        const form = tree.children[0]
        expect(form.type).toEqual('form')

        const col2 = form.children[0].children[4]

        const nameInput = col2.children[0].children[0]
        expect(nameInput.props).toHaveProperty('id', 'nameInput' )
        expect(nameInput.props).toHaveProperty('type', 'text' )
        expect(nameInput.props).toHaveProperty('name', 'name' )

        const ageInput = col2.children[1].children[0]
        expect(ageInput.props).toHaveProperty('id', 'ageInput' )
        expect(ageInput.props).toHaveProperty('type', 'number' )
        expect(ageInput.props).toHaveProperty('name', 'age' )
    })

    it('functions getResponse, getError and redux functions setItems, setName, setAge ' +
        'should be invoked on componentDidMount', () => {
        configure({ adapter: new Adapter() });

        let mockSetItems = jest.fn()
        let mockSetName = jest.fn()
        let mockSetAge = jest.fn()

        const getResponse = jest.spyOn(NewOrderStep1.prototype, 'getResponse');
        const getError = jest.spyOn(NewOrderStep1.prototype, 'getError');

        const component = shallow(
            <NewOrderStep1 setItems={mockSetItems} setName={mockSetName} setAge={mockSetAge}/>
        )

        component.instance().getResponse = jest.fn()
        component.instance().getError = jest.fn()
        component.update()

        expect(mockSetItems).toHaveBeenCalled()
        expect(mockSetName).toHaveBeenCalled()
        expect(mockSetAge).toHaveBeenCalled()
        expect(getResponse).toHaveBeenCalled()
        expect(getError).toHaveBeenCalled()
    })
})