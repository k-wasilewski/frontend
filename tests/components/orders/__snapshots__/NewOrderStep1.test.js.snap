// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewOrderStep1 rendering specification renders a form with two inputs 1`] = `
<div
  className="main"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form"
    >
      <p
        className="resp"
      />
      <h2>
        Nowe zamówienie
      </h2>
      <p
        className="hidden invalid"
      >
        
      </p>
      <div
        className="col1"
      >
        <p>
          Imię:
        </p>
        <p>
          Wiek:
        </p>
        <button>
          Dalej
        </button>
      </div>
      <div
        className="col2"
      >
        <p>
          <input
            id="nameInput"
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </p>
        <p>
          <input
            id="ageInput"
            name="age"
            onChange={[Function]}
            type="number"
            value=""
          />
        </p>
      </div>
    </div>
  </form>
</div>
`;
